pkg/objstore/azure/azure.go

```diff
@@ -98,27 +98,45 @@ func (b *Bucket) Iter(ctx context.Context, dir string, f func(string) error) err
 		prefix += DirDelim
 	}
 
-	list, err := b.containerURL.ListBlobsHierarchySegment(ctx, blob.Marker{}, DirDelim, blob.ListBlobsSegmentOptions{
+	lbs_options := blob.ListBlobsSegmentOptions{
 		Prefix: prefix,
-	})
-	if err != nil {
-		return errors.Wrapf(err, "cannot list blobs in directory %s", dir)
 	}
-	var listNames []string
 
-	for _, blob := range list.Segment.BlobItems {
-		listNames = append(listNames, blob.Name)
-	}
+	marker := blob.Marker{}
 
-	for _, blobPrefix := range list.Segment.BlobPrefixes {
-		listNames = append(listNames, blobPrefix.Name)
-	}
+	for {
+		list, err := b.containerURL.ListBlobsHierarchySegment(ctx, marker, DirDelim, lbs_options)
 
-	for _, name := range listNames {
-		if err := f(name); err != nil {
-			return err
+		if err != nil {
+			return errors.Wrapf(err, "cannot list blobs in directory %s", dir)
 		}
+
+		marker = list.NextMarker
+
+		var listNames []string
+
+		for _, blob := range list.Segment.BlobItems {
+			listNames = append(listNames, blob.Name)
+		}
+
+		for _, blobPrefix := range list.Segment.BlobPrefixes {
+			listNames = append(listNames, blobPrefix.Name)
+		}
+
+		for _, name := range listNames {
+			if err := f(name); err != nil {
+				return err
+			}
+		}
+
+		// Continue iterating if we are not done.
+		if !marker.NotDone() {
+			break
+		}
+
+		level.Info(b.logger).Log("msg", "Continuing iteration")
```

Comment:
Ah.. but you know that this flow is used for other operations as well? So with that metadata it would be as easy as changing to debug level to see more info for azure provider for compactor for example?

For ls command, sure this would be enough (:

Reply:
Also `contuining iteration` -> `requesting new page for list blobs`?

Reply:
Ok